### DISCUSSION TITLE
PocketHost 0.4.0 released

### DISCUSSION THREAD

**Discussion by benallfree** [pockethost/pockethost#55] [Show and tell]
[https://pockethost.io](https://pockethost.io/), the zero-config hosting platform for PocketBase, has reached v0.4.0. Continued thanks to @brewhousedigital for the remarkable frontend support and of course to @ganigeorgiev for PocketBase.

In this release, we've added several important features:

## PocketBase 0.8 support

PocketHost now supports [PocketBase v0.8](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc1).

## Platforms

In this release, PocketHost introduces the concept of *platforms* following [a discussion about how to manage upgrades and migrations](https://github.com/pocketbase/pocketbase/discussions/880). A platform is a set of PocketBase versions that are backward and forward compatible with one another. Currently, two platforms exist:

* Ermine - 0.7.x
* Lollipop - 0.8.x

Existing projects are running Ermine. New projects will run Lollipop. 

The pockethost.io backend and frontend have been upgraded to Lollipop.

In a future pockethost.io release, users will be able to:
* migrate existing instances and data forward and backward across platforms
* lock the instance to a specific PocketBase version
* lock to `latest` to allow instances to use the latest PocketBase version in the selected platform
* back up and restore instance data

**(nobody) Q: Where are the platform names coming from?**  
A: We compiled a long list of gopher names.

## Verified Accounts

Your account must now be verified before an instance will run. 

## OpenGraph 

Sharing pockethost.io links will now look good on ~Elon~Twitter, Facebook, et al. Thanks @brewhousedigital for this one!

## Darkmode enhancements

There was a slight flicker of day mode on page reload. This caused temporary night blindness in testers. The site now loads in Dark Mode according to cookie settings without the temporary flash of day mode.

## Closing in on cloud functions

https://github.com/benallfree/pbscript/ has been archived and is now part of the pockethost repo. In future iterations, we'll release custom PocketBase builds that support authoring PocketBase hooks in TS/JS.

## Stats

We'll have more insight into active usage stats once #21 is implemented.

* Instances: 113
* Users: 95
* Hardware:  2vCPU / 4 GB Memory / 25 GB Disk / SFO3

**Comment by DobroslavRadosavljevic** [comment]
Hey! Nice work! I have a question regarding the future of this project. How is it with pricing, etc.? Right now, it's free from what I know. So what are the plans for the future? Thanks!

**Reply by benallfree** [reply]
This is under discussion in #44. No solid plans yet, except the intent would be to base it on minutes of runtime. 

Unless you’re using real-time features, it’s unlikely you’d ever run out of minutes on a small or medium project.
